imagenet/main.py

```diff
@@ -250,14 +283,17 @@ def main_worker(gpu, ngpus_per_node, args):
         best_acc1 = max(acc1, best_acc1)
 
         if not args.multiprocessing_distributed or (args.multiprocessing_distributed
-                and args.rank % ngpus_per_node == 0):
+                and args.rank == 0):
+            # To support auto resizing, in theory we should check file existence here,
+            # to decide whether we should save checkpoint. For ImageNet sample, we save 
+            # checkpoint for every epoch, so it's fine we don't check the file here. 
```

Comment:
how about just support epoch level elastic as the first version?
we can support step level after we figured out how to use DistributedSampler properly and save dataset info into ckpt.